airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -80,6 +81,44 @@ def ensure_at_least_one_state_emitted(self) -> None:
         pass
 
 
+class FinalStateCursor(Cursor):
```

Comment:
Honestly, I think I would make this the `NoopCursor` as in the current state with the new logic of "we always have to emit a state", `NoopCursor` should never be used. This is a breaking change as it requires to change the constructor but it feels like `NoopCursor` from this PR is just invalid as it does not respect basic domain responsibilities. The annoying part in this is updating the tests but it seems fair that our tests reflect real prod usages.

Reply:
True, I think removing the NoopCursor makes sense. 

>  This is a breaking change as it requires to change the constructor
What's the implication here? (I'm not super familiar with breaking changes to the CDK) It's mostly that when we upgrade the CDK version of a given connector, this has the potential to break if they're using it in a certain way? If someone is not using the NoOpCursor explicitly, does it affect them for us to change the default under the hood?

Reply:
Honestly, we marked most of the concurrent interface as experimental so I wouldn't mind the breaking change. I think it would only involve updating two lines (one for stripe and one for salesforce).

Reply:
@maxi297 i think its fair to just yank it. I'm probably being more cautious than needed here